server/lib/routes/otp.js

```diff
@@ -75,23 +76,27 @@ async function otpRoutes(server) {
               // update due to object creation
               return
             }
-            completeRequest(null, update.get('otp'))
+            completeRequest(null, update.get('otp'), requestObj)
           },
           (error) => {
             log.error(error.message)
-            return reply.code(500).send(error.message)
+            return reply.internalServerError('An unexpected error occured')
           }
         )
 
-        // fire the notification to all available subscription
         push.send({
-          subscriptions: subscriptions.docs,
+          subscription: subscription.docs[0].data(),
```

Comment:
I'm still somewhat surprised that we're using a query which returns possibly multiple docs when all we need is to look up 1 doc by its primary key

Reply:
indeed there is a way to do that https://github.com/nearform/optic/pull/46/files#diff-32a64babda63d3014c589afe5b130e3e927456f69b66a397a0f30715d570eb1aR110

Reply:
I have corrected this now